changelog.d/10245.feature

```diff
@@ -0,0 +1 @@
+Make historical events discoverable from backfill for servers without any scrollback history (part of MSC2716).
```

Comment:
@erikjohnston I'm curious about your general sense for how I am tackling this bit so far?

For easy reference, here is what this PR does so far:

> Make historical events discoverable from backfill for servers without any scrollback history. This does not cover "marker" events or servers which already have all of the scrollback(no backwards extremeties to pull from)

I've got it working for auto-generated insertion event scenario. But still need to get the [Complement test](https://github.com/matrix-org/complement/pull/68) passing for when the insertion event is already in the "live" DAG and we add history onto that.

Reply:
Thanks for the review pass @erikjohnston 🦘. I've run into a few head scratchers around [securing who can send these new events](https://github.com/matrix-org/synapse/pull/10245#discussion_r669979503) and the [`state_groups` bug](https://github.com/matrix-org/synapse/pull/10245#discussion_r670145534) and could use some thoughts!